radixdlt-core/radixdlt/src/test/java/com/radixdlt/stateir/StateIRSerializationTest.java

```diff
@@ -0,0 +1,282 @@
+import com.radixdlt.store.berkeley.BerkeleyLedgerEntryStore;
+import com.radixdlt.utils.PrivateKeys;
+import com.radixdlt.utils.UInt256;
+import java.io.ByteArrayInputStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.IntStream;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+public final class StateIRSerializationTest {
+  private static final ECKeyPair VALIDATOR_KEY = PrivateKeys.ofNumeric(1);
+
+  @Rule public TemporaryFolder folder = new TemporaryFolder();
+
+  @Inject private RadixEngine<LedgerAndBFTProof> radixEngine;
+
+  @Inject private BerkeleyLedgerEntryStore ledgerEntryStore;
+
+  // FIXME: Hack, need this in order to cause provider for genesis to be stored
+  @Inject @LastStoredProof private LedgerProof ledgerProof;
+
+  @Inject private Forks forks;
+
+  private final Random rnd = new Random();
+
+  private Injector createInjector() {
+    return Guice.createInjector(
+        MempoolConfig.asModule(1000, 10),
+        new MainnetForksModule(),
+        new RadixEngineForksLatestOnlyModule(
+            RERulesConfig.testingDefault()
+                .overrideFeeTable(
+                    FeeTable.create(
+                        Amount.zero(), Map.of(TokenResource.class, Amount.ofTokens(1))))),
+        new ForksModule(),
+        new SingleNodeAndPeersDeterministicNetworkModule(VALIDATOR_KEY, 0),
+        new MockedGenesisModule(
+            Set.of(VALIDATOR_KEY.getPublicKey()),
+            Amount.ofTokens(100_000_000_000L),
+            Amount.ofTokens(100L)),
+        new AbstractModule() {
+          @Override
+          protected void configure() {
+            bindConstant()
+                .annotatedWith(DatabaseLocation.class)
+                .to(folder.getRoot().getAbsolutePath());
+          }
+        });
+  }
+
+  @Test
+  public void
+      it_should_successfully_construct_and_then_serialize_and_deserialize_the_intermediate_state()
+          throws Exception {
+    createInjector().injectMembers(this);
+
+    final var numAccounts = 100;
+    final var numTokens = 10;
+    final var transfersPerToken = 100;
+
+    final var accounts =
+        IntStream.range(0, numAccounts)
+            .mapToObj(unused -> REAddr.ofPubKeyAccount(ECKeyPair.generateNew().getPublicKey()))
+            .toList();
+
+    for (REAddr accountAddr : accounts) {
+      transfer1Rad(VALIDATOR_KEY, accountAddr);
+    }
+
+    for (int i = 0; i < numTokens; i++) {
+      final var tokenAddr = createMutableToken(VALIDATOR_KEY, UInt256.from(10_000_000_000L));
+      final var accountsForTransfer = new ArrayList<REAddr>();
+      for (int j = 0; j < transfersPerToken; j++) {
+        accountsForTransfer.add(accounts.get(rnd.nextInt(accounts.size())));
+      }
+
+      transferRandomAmtOfTokensToAccounts(VALIDATOR_KEY, tokenAddr, accountsForTransfer, 1000);
+    }
```

Comment:
Same here. It could become:

`private void makeRandomTransfer(int numTokens, int transfersPerToken, List<REAddr> accounts) throws TxBuilderException, RadixEngineException `